lib/src/rules/invalid_case_patterns.dart

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) 2023, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+
+import '../analyzer.dart';
+
+const _desc = r'Use case expressions that are valid in Dart 3.0.';
+
+/// todo(pq): add details or link out to a doc?
+const _details = r'''
+Some case expressions that are valid in Dart 2.19 and below will become an error
+or have changed semantics when a library is upgraded to 3.0. This lint flags
+those expressions in order to ease migration to Dart 3.0.
+''';
+
+class InvalidCasePatterns extends LintRule {
+  static const LintCode code = LintCode(
+      'invalid_case_patterns', 'Invalid case pattern.',
+      // todo(pq): can we have a doc link here?
```

Comment:
I'm guessing that you mean "can we link directly to some doc other than the one generated from the `_details`. In which case the answer is that it isn't currently supported.

I think the link to the other doc should be in the `_details`. There will automatically be a link in the IDE to the doc page generated from the `_details`. That might seem like a bit of an indirection, but it's also more consistent with the way the rest of the IDE links work, and I think that consistency trumps the extra indirection.